File: Selenium/Activities/Activity8.java
package activity;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity8 {

	public static void main(String[] args) throws InterruptedException {
		
		WebDriver driver = new ChromeDriver();
		Actions action = new Actions(driver);
		
		driver.get("https://training-support.net/webelements/mouse-events");
		
		System.out.println("Title of the page: " + driver.getTitle());
		
		WebElement cargoLockBtn = driver.findElement(By.xpath("//h1[.='Cargo.lock']"));
		WebElement cargoTomlBtn = driver.findElement(By.xpath("//h1[.='Cargo.toml']"));
		WebElement srcBtn = driver.findElement(By.xpath("//h1[.='src']"));
		WebElement targetBtn = driver.findElement(By.xpath("//h1[.='target']"));
		WebElement result = driver.findElement(By.cssSelector("#result"));
		
		action.click(cargoLockBtn).moveToElement(cargoTomlBtn).click().build().perform();
		
		System.out.println("Result Text: " + result.getText());
		
		action.doubleClick(srcBtn).contextClick(targetBtn).click().build().perform();
		
		WebElement open = driver.findElement(By.xpath("//button[@class = 'flex w-full p-2']//span[.='Open']"));
		
		action.click(open).perform();
		Thread.sleep(2000);
		
		System.out.println("Result Text: " + result.getText());
		
		driver.quit();

	}

}

File: Selenium/Activities/activity_8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver import ActionChains

with webdriver.Firefox() as driver:
	
    actions = ActionChains(driver)
	
    driver.get("https://training-support.net/webelements/mouse-events")
    
    print("Title of the page: ", driver.title)

    cargo_lock_btn = driver.find_element(By.XPATH, "//h1[.='Cargo.lock']")
    cargo_toml_btn = driver.find_element(By.XPATH, "//h1[.='Cargo.toml']")
    src_btn = driver.find_element(By.XPATH, "//h1[.='src']")
    target_btn = driver.find_element(By.XPATH, "//h1[.='target']")
    result = driver.find_element(By.CSS_SELECTOR, "#result")
	
    actions.click(cargo_lock_btn).move_to_element(cargo_toml_btn).click().perform()
		
    print("Result Text: ", result.text)
		
    actions.double_click(src_btn).context_click(target_btn).click().perform()
		
    open = driver.find_element(By.XPATH, "//button[@class = 'flex w-full p-2']//span[.='Open']")
		
    actions.click(open).pause(2).perform()
		
    print("Result Text: ", result.text)
		
    driver.quit()